### ISSUE TITLE
Changing ENV's from values to valuesfrom throws error

### ISSUE THREAD

**Issue by Spazzy757** [kbst/terraform-provider-kustomization#86]
## Problem
When Changing an environment from using a literal value to reading a value from a secret and error seems to occur showing that you can not have both set

## Example:

Before:
```yaml
...
env:
- name: KEYCLOAK_FRONTEND_URL
  value: example.com
...
```

After:
```yaml
...
env:
- name: KEYCLOAK_FRONTEND_URL
  valueFrom:
      secretKeyRef:
          name: keycloak-config
          key: KEYCLOAK_FRONTEND_URL
...
```

Causes the following error:

```log
Error: github.com/kbst/terraform-provider-kustomize/kustomize.kustomizationResourceDiff: apiVersion: "apps/v1", kind: "Deployment", namespace: "keycloak" name: "keycloak": patch failed 'application/strategic-merge-patch+json': Deployment.apps "keycloak" is invalid: spec.template.spec.containers[0].env[0].valueFrom: Invalid value: "": may not be specified when `value` is not empty

  on .terraform/modules/eks_zero/common/cluster_services/main.tf line 16, in resource "kustomization_resource" "current":
  16: resource "kustomization_resource" "current" {
```

The same issue seems to happpen when you change a container port on a service or a pod that hase a name e.g `name: http`

**Comment by pst** [comment]
Thanks for reporting. I found the issue. The provider tries `StrategicMergePatch` first and is supposed to fall back to `MergePatch` when `StrategicMergePatch` is not an option. But it currently will only try `MergePatch` if the error returned is `UnsupportedMediaType`. However, in this case, the API returns `Invalid`. If the API returns `Invalid` but we have only tried a `StrategicMergePatch` so far, we have to still try `MergePatch`.

I will release a fix today.
